crates/matrix-sdk-crypto/src/types/events/mod.rs

```diff
@@ -18,16 +18,27 @@
 //! types. Once deserialized they aim to zeroize all the secret material once
 //! the type is dropped.
 
+pub mod room;
 pub mod room_key;
 pub mod secret_send;
 mod to_device;
 
+use ruma::serde::Raw;
 pub use to_device::{ToDeviceCustomEvent, ToDeviceEvent, ToDeviceEvents};
 
 /// A trait for event contents to define their event type.
 pub trait EventType {
+    /// The event type of the event content.
+    const EVENT_TYPE: &'static str;
+
     /// Get the event type of the event content.
-    fn event_type(&self) -> &str;
+    fn event_type(&self) -> &'static str {
```

Comment:
Is this just kept for backwards compatibility, or is it actually legal for this function to return something else than the constant in implementations of this trait?

Reply:
I don't think it's legal to return anything else, no.

Reply:
Maybe have a comment saying as much then, or if it's not too involved, just remove the `fn`.